questionpy_sdk/webserver/question_ui/__init__.py

```diff
@@ -76,10 +76,24 @@ def _set_element_value(element: etree._Element, value: str, name: str, xpath: et
         element.set("value", value)
 
 
+def _check_shuffled_index_is_in_nested_shuffle_contents(index_element: etree._Element) -> bool:
+    ancestor = index_element.getparent()
+    while ancestor is not None and ancestor != index_element:
+        if f"{{{_QPY_NAMESPACE}}}shuffle-contents" in ancestor.attrib:
+            return True
+        ancestor = ancestor.getparent()
+    return False
```

Comment:
Da habe ich komischen Code geschrieben, den du jetzt (korrekt) übernommen hast.
Das `ancestor != index_element` ist sinnlos, gemeint war (in PHP) eigentlich `$ancestor !== $element`, also dass bei erreichen des aktuell behandelten `shuffle-contents`-Elements aufgehört wird. Ich muss mich damals vertippt haben. Dass es trotzdem funktioniert liegt daran, dass das aktuell behandelte `shuffle-contents`-Attribut und alle höher liegenderen entfernt wurden.

Ich passe das im Plugin mal an und würde vorschlagen, dass du es hier auch korrigierst.